README.md

```diff
@@ -4,204 +4,430 @@ Daniel Ehrenberg, Igalia. Stage 3
 
 Thanks for help and feedback on this effort from Brendan Eich, Waldemar Horwat, Jaro Sevcik, Benedikt Meurer, Michael Saboff, Adam Klein and others.
 
-## Code samples:
+## Contents
+1. [What Is It?](#what-is-it)
+2. [How Does It Work?](#how-does-it-work)
+	- [Syntax](#syntax)
+	- [Example: Calculating Primes](#example-calculating-primes)
+	- [Operators](#operators)
+	- [Comparisons](#comparisons)
+	- [Conditionals](#conditionals)
+	- [Other API Notes](#other-api-notes)
+3. [Gotchas & Exceptions](#gotchas--exceptions)
+	- [Interoperation with `Number` and `String`](#interoperation-with-number-and-string)
+	- [Rounding](#rounding)
+	- [Other Exceptions](#other-exceptions)
+	- [Usage Recommendation](#usage-recommendation)
+4. [About the Proposal](#about-the-proposal)
+	- [Motivation, Or Why Do We Need Such Big Numbers?](#motivation-why-do-we-need-such-big-numbers)
+	- [Design Philosophy, Or Why Is This Like This?](#design-goals-or-why-is-this-like-this)
+	- [State of the Proposal](#state-of-the-proposal)
+
+
+## What Is It?
+
+`BigInt` is a new primitive that provides a way to represent whole numbers larger than 2<sup>53</sup>, which is the largest number Javascript can reliably represent with the `Number` primitive.
 
-Find the nth prime:
+```js
+const x = Number.MAX_SAFE_INTEGER;
+// ↪ 9007199254740991, this is 1 less than 2^53
+
+const y = x + 1;
+// ↪ 9007199254740992, ok, checks out
+
+const z = x + 2
+// ↪ 9007199254740992, wait, that’s the same as above!
+```
+
+[Learn more about how numbers are represented in Javascript in the slides from Daniel's talk at JSConfEU.](https://docs.google.com/presentation/d/1apPbAiv_-mJF35P31IjaII8UA6TwSynCA_zhfDEmgOE/edit#slide=id.g38a1897a56_0_97) 
+
+## How Does It Work?
+
+The following sections show `BigInt` in action. A number have been influenced by or taken outright from [Mathias Bynens's Big int v8 update, which incudes more detail that this page.](https://developers.google.com/web/updates/2018/05/bigint)
+
+### Syntax
+
+A `BigInt` is created by appending `n` to the end of the integer or by calling the constructor.
+
+```js
+
+const theBiggestInt = 9007199254740991n;
+
+const alsoHuge = BigInt(9007199254740991);
+// ↪ 9007199254740991n
+
+const hugeButString = BigInt('9007199254740991');
+// ↪ 9007199254740991n
+
+```
+
+### Example: Calculating Primes
 
 ```js
-// Takes a BigInt as an argument and returns a BigInt
+// Takes a Number or BigInt as an argument and returns a BigInt
 function nthPrime(nth) {
+  
+  function makeBig(int) {
```

Comment:
Seemed like a nice opportunity to provide a test and conversion example as well. Is this how you'd recommend doing it?

Reply:
I wouldn't recommend doing conversion; I think it would be best to expect a BigInt as an argument, and leave the conversion to the caller.